test/inductor/test_fp8.py

```diff
@@ -56,10 +44,54 @@ def fp8_matmul_unwrapped(x):
             )
             return output
 
+        compiled_fp8_matmul = torch.compile(
+            fp8_matmul_unwrapped, backend="inductor", dynamic=True
+        )
+
+        x_shape = (16, 16)
+        x = torch.rand(*x_shape, device="cuda", dtype=dtype).to(torch.float8_e4m3fn)
+        y_fp8 = compiled_fp8_matmul(x)
+
+        x_shape = (15, 16)
         x = torch.rand(*x_shape, device="cuda", dtype=dtype).to(torch.float8_e4m3fn)
-        compiled_fp8_matmul = torch.compile(fp8_matmul_unwrapped, backend="inductor")
         y_fp8 = compiled_fp8_matmul(x)
 
+    @unittest.skipIf(TEST_WITH_ROCM, "FP8 is not supported on ROCM")
+    @unittest.skipIf(not SM90OrLater, "FP8 is only supported on H100+")
+    @parametrize("dtype", (torch.float16, torch.bfloat16, torch.float))
+    def test_valid_cast(self, dtype: torch.dtype):
+        def fp8_cast(x):
+            y0 = x.to(dtype=torch.float8_e4m3fn).to(dtype)
+            y1 = x.to(dtype=torch.float8_e5m2).to(dtype)
+            return y0, y1
+
+        compiled_fp8_cast = torch.compile(fp8_cast, backend="inductor", dynamic=True)
+
+        x_shape = (16, 16, 16)
+        x = torch.rand(*x_shape, device="cuda", dtype=dtype)
+        y0_fp8, y1_fp8 = compiled_fp8_cast(x)
+
+        torch.testing.assert_close(y0_fp8, x, rtol=5e-1, atol=5e-1)
+        torch.testing.assert_close(y1_fp8, x, rtol=5e-1, atol=5e-1)
+
+    @unittest.skipIf(TEST_WITH_ROCM, "FP8 is not supported on ROCM")
+    @unittest.skipIf(not SM90OrLater, "FP8 is only supported on H100+")
+    def test_bad_cast(self):
+        def fp8_cast(x, dtype):
+            return x.to(dtype=dtype)
+
+        compiled_fp8_cast = torch.compile(fp8_cast, backend="inductor", dynamic=True)
+
+        x_shape = (16, 16, 16)
+
+        with self.assertRaises(Exception):
```

Comment:
Would it make sense to be more specific here re. the `Exception` type (and maybe text)? And below.